nilearn/glm/first_level/first_level.py

```diff
@@ -1292,7 +1292,14 @@ def first_level_from_bids(
 
     dataset_path = Path(dataset_path).absolute()
 
-    kwargs_load_confounds = _check_kwargs_load_confounds(**kwargs)
+    kwargs_load_confounds, remaining_kwargs = _check_kwargs_load_confounds(
+        **kwargs
+    )
+
+    if len(remaining_kwargs) > 0:
+        raise RuntimeError(
+            f"Some keyword arguments were unused: {remaining_kwargs}"
```

Comment:
I think the error message could be made more specifc that something is wrong with confounds kwargs. For example:

```suggestion
            f"Unknown keyword arguments. Keyword arguments should start with `confounds_` prefix: {remaining_kwargs}"
```